mosh hardening flags conflict with Ubuntu Precise hardening flags

mosh (current git master) didn't build on the Ubuntu precise PPA builder, because Ubuntu's hardening flags work poorly with our hardening flags. It built fine on the other Ubuntu releases.

```
g++ -DHAVE_CONFIG_H -I. -I../..  -I./../util  -D_FORTIFY_SOURCE=2 -Wall -Werror -Wextra -pedantic -Wno-long-long -Weffc++ -fno-strict-overflow -D_FORTIFY_SOURCE=2 -fstack-protector-all -Wstack-protector --param ssp-buffer-size=1 -fPIE -fno-default-inline -pipe -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Wformat-security -c -o terminaldispatcher.o terminaldispatcher.cc
terminaldispatcher.cc: In member function 'void Terminal::Dispatcher::dispatch(Terminal::Function_Type, const Parser::Action*, Terminal::Framebuffer*)':
terminaldispatcher.cc:173:6: error: stack protector not protecting function: all local arrays are less than 4 bytes long [-Werror=stack-protector]
cc1plus: all warnings being treated as errors
make[4]: *** [terminaldispatcher.o] Error 1
```

https://launchpadlibrarian.net/102270658/buildlog_ubuntu-precise-amd64.mosh_1.1.94-0~684~precise1_FAILEDTOBUILD.txt.gz

--- a/debian/rules
+++ b/debian/rules
@@ -9,6 +9,10 @@
 # Uncomment this to turn on verbose mode.
 #export DH_VERBOSE=1
 
+# Through Autoconf we set hardening flags that are actually more aggressive
+# than the Ubuntu defaults, but they conflict with same.
+export DEB_BUILD_HARDENING_STACKPROTECTOR=0
+
 %:
 	dh $@ --with autoreconf
 

